integration-tests/src/test/resources/apps/cdtservlet/src/application/CdtTxServlet.java

```diff
@@ -0,0 +1,246 @@
+
+      //Get the context from domain1. This is where JMS Queue is configured
+      ctx = new InitialContext(h1);
+      System.out.println("Got Local InitialContext [" + ctx + "]");
+      out.println("Got Local InitialContext [" + ctx + "]");
+
+      ConnectionFactory qcf = (ConnectionFactory) ctx.lookup("weblogic.jms.XAConnectionFactory");
+      d = (Destination) ctx.lookup("jms/testCdtUniformQueue");
+      System.out.println("Got Local JMS Destination " + d);
+      out.println("Got Local JMS Destination " + d);
+      context = qcf.createContext();
+
+      //Get context from domain2. This is where JDBC DS is configured
+      out.println("Getting Remote Context from " + domain2Url);
+      Hashtable h = new Hashtable();
+      h.put(Context.INITIAL_CONTEXT_FACTORY,
+          "weblogic.jndi.WLInitialContextFactory");
+      h.put(Context.PROVIDER_URL, domain2Url);
+
+      Context ctx2 = new InitialContext(h);
+      System.out.println("Got Remote InitialContext [" + ctx2 + "]");
+      out.println("Got Remote InitialContext [" + ctx2 + "]");
+
+      javax.sql.DataSource ds = (javax.sql.DataSource) ctx2.lookup("jdbc/TestCdtDataSource");
+      System.out.println("Got ds from context - " + ds);
+      out.println("Got ds from context - " + ds);
+      conn = ds.getConnection();
+      System.out.println("Got connection to datasource - " + conn);
+      out.println("Got connection to datasource - " + conn);
+
+      //create a table in the DB
+      createTable(conn, tableName);
+
+      try {
+        tm.begin();
+        Transaction tx = (Transaction) tm.getTransaction();
+        Xid xid = tx.getXID();
+        System.out.println("xid=" + xid);
+        tx.setProperty("TMAfterTLogBeforeCommitExit", Boolean.TRUE);
+        
+        Connection conn1 = ds.getConnection();
+        System.out.println("Got connection to datasource - " + conn1);
+        out.println("Got connection to datasource - " + conn1);
+        insertData(conn1, tableName);
+        System.out.println("CdTXServlet: Inserted data");
+        out.println("CdTXServlet: Inserted data");
+        context.createProducer().send(d, "(D1) A Transcated JMS Message");
+        System.out.println("CdTXServlet: sent message to jms queue");
+        out.println("CdTXServlet: sent message to jms queue");
+
+        System.out.println("tx.toString() = " + tx.toString());
+        out.println("tx.toString() = " + tx.toString());
+
+        tm.commit();
+      } catch (SystemException se) {
+        if (!(se.toString().contains("weblogic.rjvm.PeerGoneException"))) {
+          throw se;
+        }
+        System.out.println("CdTXServlet: Got PeerGoneException " + se);
+        out.println("CdTXServlet:Exception got - " + se.toString());
+        out.println("CdTXServlet: got PeerGoneException as expected");
+      }
+
+      out.println("Message Delivered in Distributed Transcation");
+
+      sleep(30);
+
+      String body = context.createConsumer(d).receiveBody(String.class);
```

Comment:
Check the message  body , to see if the body contains the same message string.

Reply:
fixed.